src/controllers/auth.js

```diff
@@ -1,11 +1,92 @@
+const jwt = require('jsonwebtoken');
+
+const bcrypt = require('bcrypt');
+
+const saltRounds = 10;
+
+const User = require('../models/user');
+
 module.exports = {
-  signInUser: (req, res) => {
-    res.send('You successfully signed in');
+  signInUser: async (req, res) => {
+    const { username, email, password, rememberMe } = req.body;
+    let cookieAge = 24 * 3600;
+    try {
+      const user = await User.findOne({
+        $or: [{ email: email }, { username: username }],
+      });
+      /* eslint object-shorthand: ["warn"] */
+      if (!user) throw new Error('Wrong username');
+      const validPassword = await bcrypt.compare(password, user.password_hash);
+      if (!validPassword) throw new Error('Wrong password');
+
+      const payload = {
+        username,
+        email: user.email,
+        firstName: user.firstName,
+        lastName: user.lastName,
+      };
+
+      if (rememberMe) {
+        cookieAge = 14 * 24 * 3600;
```

Comment:
You can rewrite this and add a comment as: `cookieAge = 14 * cookieAge; // Remember me setting extends cookie expiry time to 2 weeks`